compiler/tests/evm_contracts.rs

```diff
@@ -0,0 +1,321 @@
+        )
+    })
+}
+
+#[test]
+fn multi_param() {
+    with_executor(&|mut executor| {
+        let harness = deploy_contract(&mut executor, "multi_param.vy", "Foo");
+
+        harness.test_function(
+            &mut executor,
+            "bar",
+            vec![u256_token(4), u256_token(42), u256_token(420)],
+            Some(u256_array_token(vec![4, 42, 420])),
+        )
+    })
+}
+
+#[test]
+fn u256_u256_map() {
+    with_executor(&|mut executor| {
+        let harness = deploy_contract(&mut executor, "u256_u256_map.vy", "Foo");
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![u256_token(4), u256_token(42)],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![u256_token(420), u256_token(12)],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "read_bar",
+            vec![u256_token(4)],
+            Some(u256_token(42)),
+        );
+
+        harness.test_function(
+            &mut executor,
+            "read_bar",
+            vec![u256_token(420)],
+            Some(u256_token(12)),
+        );
+    })
+}
+
+#[test]
+fn address_bytes10_map() {
+    with_executor(&|mut executor| {
+        let harness = deploy_contract(&mut executor, "address_bytes10_map.vy", "Foo");
+
+        let address1 = address_token("0000000000000000000000000000000000000001");
+        let bytes1 = bytes_token("ten bytes1");
+
+        let address2 = address_token("0000000000000000000000000000000000000002");
+        let bytes2 = bytes_token("ten bytes2");
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![address1.clone(), bytes1.clone()],
+            None,
+        );
+
+        harness.test_function(
+            &mut executor,
+            "write_bar",
+            vec![address2.clone(), bytes2.clone()],
+            None,
+        );
+
+        harness.test_function(&mut executor, "read_bar", vec![address1], Some(bytes1));
+
+        harness.test_function(&mut executor, "read_bar", vec![address2], Some(bytes2));
+    })
+}
+
+#[test]
+fn guest_book() {
+    with_executor(&|mut executor| {
+        let mut harness = deploy_contract(&mut executor, "guest_book.vy", "GuestBook");
+
+        let sender = address_token("1234000000000000000000000000000000005678");
+        let bytes = bytes_token(
+            iter::repeat("ten bytes.")
+                .take(10)
+                .collect::<String>()
+                .as_str(),
+        );
+
+        harness.caller = sender.clone().to_address().unwrap();
+
+        harness.test_function(&mut executor, "sign", vec![bytes.clone()], None);
+
+        harness.test_function(&mut executor, "get_msg", vec![sender], Some(bytes.clone()));
+
+        // Destruct the executor
+        harness.event_emitted(executor, "Signed", vec![bytes]);
```

Comment:
what exactly does `Destruct` mean here? i'm assuming just bc we are leaving the `with_executor` block but this comment reads like it is attached to the event check

Reply:
I guess [deconstruct](https://docs.rs/evm/0.16.1/evm/executor/struct.StackExecutor.html#method.deconstruct) might be a better term...

The only way you can get events from the executer is by passing self by value into deconstruct, which isn't ideal. The comment isn't very useful tbh.